enhancements/storage/csi-migration.md

```diff
@@ -0,0 +1,218 @@
+
+1. Tech Preview in 4.9:
+
+* Update the carry-patch in Attach Detach Controller to force-enable the CSI Migration for EBS.
+
+2. GA in 4.10:
+
+* Update the carry-patch in Attach Detach Controller to **not** force-enable the CSI Migration for EBS.
+
+Once CSI Migration is GA for all storage plugins have been completely migrated to CSI, we can:
+
+* Remove the carry-patch from OCP.
+* Remove the `CSIMigration` _FeatureSet_ from openshift/api.
+
+
+## Implementation History
+
+Main events only, this is not a faithful history.
+
+2020-07-29: Initial enhancement draft.
+2021-01-28: Re-worked proposal to use 2 _FeatureSets_ with manual application by the user.
+2021-03-05: Re-worked proposal to use carry-patch instead. Moved previous approach to "Alternatives" for reference.
+
+## Alternatives
+
+We have considered this alternative approach. It is NOT our preferable approach because it introduces many shortcomings:
+
+* We put on the user the responsability of applying the *FeatureSets* in the correct order, which is very error-prone.
+* We need to create 2 extra *FeatureSets* as opposed to 1.
+* We need to patch MCO so that it ignores one of these *FeatureSets*.
+* Once GA, downgrades to a previous version with CSI Migration disabled by default would break because OCP downgrades the control-plane before nodes.
+
+### Tech Preview
+
+1. Create two new FeatureSets: `CSIMigrationNode` and `CSIMigrationControlPlane`. Both *FeatureSets* will enable all CSI Migration feature flags.
+   * The machine-config-operator (MCO) will **ignore** the `CSIMigrationControlPlane` **FeatureSet**.
+   * On the other hand, all operators will react to the `CSIMigrationNode` *FeatureSet*, including control-plane operators.
+2. To enable CSI Migration for any in-tree plugin, the cluster administrator should:
+   * First, enable the CSI migration feature flags in all control-plane components by adding the `CSIMigrationControlPlane` *FeatureSet* to the `featuregates/cluster` object.
+     - With the exception of MCO, all operators will recognize this *FeatureSet* and will initialize their operands with the associated feature gates.
+   * Second, once all control-plane components have restarted, enable the CSI migration feature flags in the Kubelet:
+     - Add the `CSIMigrationNode` *FeatureSet* to the `featuresgates/cluster` object, replacing the previous value (i.e., `CSIMigrationControlPlane`).
+     - All operators will recognize the `CSIMigrationNode` *FeatureSet*, however, control-plane operators already applied the associated feature gates in the step above, so in practice only MCO will have work to do.
+   * At this point, CSI migration is fully enabled in the cluster.
+3. To disable CSI migration, the cluster administrator should perform the same steps in the opposite order:
+   * In `featuregates/cluster` object, replace the `CSIMigrationNode` *FeatureSet* by `CSIMigrationControlPlane`.
```

Comment:
This could be clearer. What about:

> In the `featuregates/cluster` object, replace the `CSIMigrationNode` *FeatureSet* with `CSIMigrationControlPlane`.

Reply:
Fixed